pkg/ccl/backupccl/schedule_exec.go

```diff
@@ -462,16 +472,17 @@ func (e *scheduledBackupExecutor) OnDrop(
 	sj *jobs.ScheduledJob,
 	txn *kv.Txn,
 	descsCol *descs.Collection,
-) error {
+) (int, error) {
 	args := &backuppb.ScheduledBackupExecutionArgs{}
 	if err := pbtypes.UnmarshalAny(sj.ExecutionArgs().Args, args); err != nil {
-		return errors.Wrap(err, "un-marshaling args")
+		return 0, errors.Wrap(err, "un-marshaling args")
 	}
 
-	if err := unlinkDependentSchedule(ctx, scheduleControllerEnv, env, txn, args); err != nil {
-		return errors.Wrap(err, "failed to unlink dependent schedule")
+	numRowsDropped, err := unlinkOrDropDependentSchedule(ctx, scheduleControllerEnv, env, txn, args)
+	if err != nil {
+		return numRowsDropped, errors.Wrap(err, "failed to unlink dependent schedule")
 	}
-	return releaseProtectedTimestamp(ctx, txn, scheduleControllerEnv.PTSProvider(),
+	return numRowsDropped, releaseProtectedTimestamp(ctx, txn, scheduleControllerEnv.PTSProvider(),
```

Comment:
we'll also need to release the PTS for the incremental schedule. Perhaps a better design for this change is:

- keep unlinkDependentScheudle as is
- pull this block of code out of unlinkDependentSchedule:
```
dependentSj, dependentArgs, err := getScheduledBackupExecutionArgsFromSchedule(ctx, env, txn,
		scheduleControllerEnv.InternalExecutor().(*sql.InternalExecutor), args.DependentScheduleID)
		```
		
	- create a seperate helper func for droping the dependent scheudle